molo/core/views.py

```diff
@@ -195,10 +195,10 @@ def get_queryset(self, **kwargs):
             articles = ArticlePage.objects.filter(
                 pk__in=articles).order_by(
                     'latest_revision_created_at')
-            return get_pages(context, articles, locale)[:count]
-        return ArticlePage.objects.descendant_of(main).filter(
+            return [tag, get_pages(context, articles, locale)[:count]]
+        return [tag, ArticlePage.objects.descendant_of(main).filter(
```

Comment:
I think it's better to return the tag in `def get_context_data` rather than lumping it in the queryset.